opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanAggregationBuilder.java

```diff
@@ -8,64 +8,68 @@
 import org.opensearch.sql.planner.logical.LogicalAggregation;
+import org.opensearch.sql.planner.logical.LogicalFilter;
+import org.opensearch.sql.planner.logical.LogicalHighlight;
+import org.opensearch.sql.planner.logical.LogicalLimit;
+import org.opensearch.sql.planner.logical.LogicalNested;
+import org.opensearch.sql.planner.logical.LogicalPaginate;
+import org.opensearch.sql.planner.logical.LogicalProject;
 import org.opensearch.sql.planner.logical.LogicalSort;
-import org.opensearch.sql.storage.TableScanOperator;
-import org.opensearch.sql.storage.read.TableScanBuilder;
 
 /**
  * Index scan builder for aggregate query used by {@link OpenSearchIndexScanBuilder} internally.
  */
-class OpenSearchIndexScanAggregationBuilder extends TableScanBuilder {
+@EqualsAndHashCode
+class OpenSearchIndexScanAggregationBuilder implements PushDownQueryBuilder {
 
   /** OpenSearch index scan to be optimized. */
-  private final OpenSearchIndexScan indexScan;
+  private final OpenSearchRequestBuilder requestBuilder;
 
   /** Aggregators pushed down. */
-  private List<NamedAggregator> aggregatorList;
+  private final List<NamedAggregator> aggregatorList;
 
   /** Grouping items pushed down. */
-  private List<NamedExpression> groupByList;
+  private final List<NamedExpression> groupByList;
 
   /** Sorting items pushed down. */
   private List<Pair<Sort.SortOption, Expression>> sortList;
 
-  /**
-   * Initialize with given index scan and perform push-down optimization later.
-   *
-   * @param indexScan index scan not fully optimized yet
-   */
-  OpenSearchIndexScanAggregationBuilder(OpenSearchIndexScan indexScan) {
-    this.indexScan = indexScan;
+
+  OpenSearchIndexScanAggregationBuilder(OpenSearchRequestBuilder requestBuilder,
+                                        LogicalAggregation aggregation) {
+    this.requestBuilder = requestBuilder;
+    aggregatorList = aggregation.getAggregatorList();
+    groupByList = aggregation.getGroupByList();
   }
 
   @Override
-  public TableScanOperator build() {
+  public OpenSearchRequestBuilder build() {
     AggregationQueryBuilder builder =
         new AggregationQueryBuilder(new DefaultExpressionSerializer());
     Pair<List<AggregationBuilder>, OpenSearchAggregationResponseParser> aggregationBuilder =
         builder.buildAggregationBuilder(aggregatorList, groupByList, sortList);
-    indexScan.getRequestBuilder().pushDownAggregation(aggregationBuilder);
-    indexScan.getRequestBuilder().pushTypeMapping(
+    requestBuilder.pushDownAggregation(aggregationBuilder);
+    requestBuilder.pushTypeMapping(
         builder.buildTypeMapping(aggregatorList, groupByList));
-    return indexScan;
+    return requestBuilder;
   }
 
   @Override
-  public boolean pushDownAggregation(LogicalAggregation aggregation) {
-    aggregatorList = aggregation.getAggregatorList();
-    groupByList = aggregation.getGroupByList();
-    return true;
+  public boolean pushDownFilter(LogicalFilter filter) {
```

Comment:
Shouldn't be there a dummy `pushDownAggregation` which just returns `true`?

Reply:
Not needed because `pushDownAggregation` does not apply to non-aggregation queries. 

For example, `pushDownLimit` is on the builder interface because both aggregate and non-aggregate queries can have a limit clause but when aggregation is encountered, query scan builder is converted to an aggregation scan builder.

Reply:
I mean if `pushDownAggregation` of a `OpenSearchIndexScanAggregationBuilder` is called it would go to default implementation which returns `false` or throws.

Reply:
`OpenSearchIndexScanAggregationBuilder` does not have `pushDownAggregation` as far as I can see.